### PR TITLE
:sparkles: Manage default values for pagination and autopage pages

### PR DESCRIPTION
**Pull request by borisschapira**
When I generate my pagination pages, they're included in my sitemap because the generated pages do not have the value Front-Matter `sitemap` to `false`.

I, therefore, propose a way to add Front-Matter values to the generated pages using the configuration.

### CODE REVIEW

**Comment by borisschapira** [comment]
I've no idea why travis failed on this one. @sverrirs, would you please have a look?

**Comment by sverrirs** [comment]
Hi Boris, thanks for the PR but did you consider using this existing functionality to identify the auto generated pages? https://github.com/sverrirs/jekyll-paginate-v2/blob/master/README-GENERATOR.md#detecting-generated-pagination-pages

**Comment by borisschapira** [comment]
@sverrirs Using this features would imply modifying all the plugins that are based on the page Front-Matter content to pass information. Giving a way to add information to the generated page Front-Matter seems like a more obvious and convenient data layer in which information can be transmitted.

**Comment by borisschapira** [comment]
@sverrirs But I don't think I have your experience on this domain so I may have missed something. If you think so and have some time for an explanation, I'd love that.

**Comment by sverrirs** [comment]
Regarding the tests, Have you tried running the rake tests locally?

File: README-GENERATOR.md [hunk 1]
```diff
@@ -100,6 +100,10 @@ pagination:
   # Without file extension
   indexpage: 'index'
 
+  # Optional, the default page values of the generated pages (e.g. 'sitemap: false')
+  # Without file extension
+  default_values: nul
```
Comment (ashmaroli) [inline README-GENERATOR.md]: `nul` is not the correct usage in English.
It should either be `null`, or `~`, or simply ` `

Comment (borisschapira) [reply README-GENERATOR.md]: You're right I'll fix that with

```suggestion
  default_values: null
```

**Comment by borisschapira** [comment]
Hi, do you need anything else for my part or it this waiting for a discussion between @ashmaroli and you regarding a more global modification?

**Comment by sverrirs** [comment]
Hi Boris,

The only thing I'm waiting for is the two review comments I made last month :) about adding documentation section describing this feature, what it does and when to use it. And perhaps consider simplifying the name of the property in the defaults :) 

Otherwise I like this change.

**Comment by ashmaroli** [comment]
@borisschapira I'm not affiliated with this project. But requesting a change nonetheless.
Can you move the "Better documentation" change into a new PR..?
That way future visitors to this PR do not have to see the huge diff..

**Comment by borisschapira** [comment]
I tried to improve the documentation, tell me what you think about this version. If I change `default_values` to `page_config`, do you think it would be clearer? 
Finally, I added a simple test on the default value. I also wanted to add a test to check that once the page was generated with a `page_config` configuration, then the generated pages would effectively expose this configuration but I don't know how to do it.

**Comment by borisschapira** [comment]
Branch is rebased.

**Pull request closed without merging**